### PR TITLE
Anonymization(dbtoolsbundle)

### PR DESCRIPTION
**Pull request by AntoineAresu**
[Ticket](link_to_trello_issue)

### CODE REVIEW

File: src/Entity/BaseUser.php [hunk 1]
```diff
@@ -21,11 +22,13 @@ abstract class BaseUser implements LegacyPasswordAuthenticatedUserInterface, Use
     /**
      * @var string
      */
+    #[Anonymize('md5')]
     protected $usernameCanonical;
 
     /**
      * @var string
      */
+    #[Anonymize('email', options: ['domain' => 'yopmail.com'])]
```
Comment (NilsRec) [inline src/Entity/BaseUser.php:31]: Could be domain "anonymized.com" to make it even clearer

File: src/Entity/User.php [hunk 2]
```diff
@@ -77,12 +79,14 @@ class User extends BaseUser implements \JsonSerializable, TwoFactorInterface, Tw
...
      */
     #[Groups(['read', 'user:read', 'v3:user:read'])]
+    #[Anonymize('md5')]
     protected $username = '';
 
     /**
      * @var string
      */
     #[Groups(['read', 'user:read', 'v3:user:read', 'v3:beneficiary:write'])]
+    #[Anonymize('email', options: ['domain' => 'yopmail.com'])]
```
Comment (NilsRec) [inline src/Entity/User.php:89]: Same thing for domain name

File: src/Entity/User.php [hunk 3]
```diff
@@ -113,21 +117,25 @@ class User extends BaseUser implements \JsonSerializable, TwoFactorInterface, Tw
...
 
     /**
      * @var string
      */
     #[Groups(['read', 'user:read', 'v3:user:read', 'v3:beneficiary:write'])]
+    #[Anonymize('fr-fr.lastname')]
     private $nom;
 
     #[Groups(['read', 'user:read', 'v3:user:read', 'v3:beneficiary:write'])]
+    #[Anonymize('date', options: ['min' => '1950-01-01 00:00:00', 'max' => '2000-01-01 00:00:00'])]
```
Comment (NilsRec) [inline src/Entity/User.php:131]: No children allowed ?

File: src/Domain/Anonymization/AnonymizationHelper.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+namespace App\Domain\Anonymization;
+
+class AnonymizationHelper
+{
+    final public const string ANONYMIZED_CONTENT = 'Ce contenu a été anonymisé pour protéger les données personnelles qui pourraient être présentes';
+    final public const string ANONYMIZED_EMAIL = 'anonymized@yopmail.com';
```
Comment (NilsRec) [inline src/Domain/Anonymization/AnonymizationHelper.php:13]: contact@anonymized.com ?

File: src/Domain/Anonymization/Anonymizer/PersonalDataNameAnonymizer.php [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+use App\Domain\Anonymization\AnonymizationHelper;
+use DbToolsBundle\PackFrFR\Anonymizer\LastNameAnonymizer;
+use MakinaCorpus\DbToolsBundle\Attribute\AsAnonymizer;
+
+#[AsAnonymizer(
+    name: 'personal_data_name',
+    pack: 'reconnect',
+    description: <<<TXT
+    Anonymize personal data name according to child entity
+    TXT
```
Comment (NilsRec) [inline src/Domain/Anonymization/Anonymizer/PersonalDataNameAnonymizer.php:14]: Why not using quotes ?

**Review by tcheymol — APPROVED** [review]

**Review by tcheymol — APPROVED** [review]

**Review by tcheymol — APPROVED** [review]

**Review by NilsRec — APPROVED** [review]

**Pull request merged**